src/main/java/com/aws/greengrass/componentmanager/ComponentManager.java

```diff
@@ -508,6 +513,9 @@ public void cleanupStaleVersions() {
             for (String compVersion : removeVersions) {
                 try {
                     ComponentIdentifier identifier = new ComponentIdentifier(compName, new Semver(compVersion));
+                    // issues #1111
+                    // delete docker image before removing the recipe file
+                    removeDockerImage(identifier);
```

Comment:
This should be done inside `componentStore.deleteComponent`. And ideally, since there's a new type of artifact to cleanup now, it will make sense to refactor [artifact deletion code here](https://github.com/leoSRM/aws-greengrass-nucleus/blob/7978cba7dc6894abe74ece967e4cca02d95c044b/src/main/java/com/aws/greengrass/componentmanager/ComponentStore.java#L250,L271) into the respective artifact downloaders (might rename downloader classes too since they would handle cleanup in addition to downloads)